### PR TITLE
Add TiFlash disaggregate architecture and S3 support

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13218
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -588,6 +588,7 @@
     - [Use TiDB to Read TiFlash Replicas](/tiflash/use-tidb-to-read-tiflash.md)
     - [Use TiSpark to Read TiFlash Replicas](/tiflash/use-tispark-to-read-tiflash.md)
     - [Use MPP Mode](/tiflash/use-tiflash-mpp-mode.md)
+    - [Disaggregated Storage and Compute Architecture and Amazon S3 Support](/tiflash/tiflash-disaggregated-and-s3.md)
```
Comment (flowbehappy) [inline TOC.md:591]: Better don't use Amazon S3 as title. S3 is more like a standard, not a specific product.

Comment (qiancai) [inline TOC.md:591]:
```suggestion
    - [Disaggregated Storage and Compute Architecture and S3 Support](/tiflash/tiflash-disaggregated-and-s3.md)
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        tiup cluster reload mycluster -R tidb
+        ```
+
+## Restrictions
+
+- TiFlash does not support in-place switching between the **disaggregated storage and compute architecture** and the **coupled storage and compute architecture**. Before switching architectures, you must remove all existing TiFlash nodes.
+- After the migration from one architecture to another, all TiFlash data needs to be replicated again.
+- Only TiFlash nodes with the same architecture are allowed in the same TiDB cluster. Two architectures cannot coexist in one cluster.
+- The disaggregated storage and compute architecture only supports object storage using the S3 API, while the coupled storage and compute architecture only supports local storage.
+- When using S3 storage, TiFlash nodes cannot obtain the keys of files not on their own nodes, so the [static encryption](/encryption-at-rest.md) feature is cannot be used.
```
Comment (qiancai) [inline tiflash/tiflash-disaggregated-and-s3.md:181]:
```suggestion
- When using S3 storage, TiFlash nodes cannot obtain the keys of files not on their own nodes, so the [Encryption at Rest](/encryption-at-rest.md) feature is cannot be used.
```

**Review by flowbehappy — APPROVED** [review]
LGTM

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    ```shell
+    "Expiration": {
+        "Days": 1
+    },
+    "Tags": [
+        {
+            "Value": "tiflash_deleted", 
+            "Key": "true"
+        }
+    ]
```
Comment (flowbehappy) [inline tiflash/tiflash-disaggregated-and-s3.md:76]:
```suggestion
"Filter": {
    "And": {
        "Tags": [
            {
                "Value": "tiflash_deleted", 
                "Key": "true"
            }
        ]
    }
}
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    - CopyObject
+    - DeleteObject
+    - ListObjectV2
+    - GetObjectTagging
+    - PutBucketLifecycle
+
+2. Add a [lifecycle](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html) to the prepared S3 bucket for cleaning up deleted data:
+
+    ```shell
+    "Expiration": {
```
Comment (flowbehappy) [inline tiflash/tiflash-disaggregated-and-s3.md:68]:
```suggestion
    "ID": "tiflash-clean",
    "Expiration": {
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+    The Write Node uses local disks (usually NVMe SSDs) to cache the latest written data to avoid excessive use of memory.
+
+    The Write Node has a faster scaling speed than the TiFlash node in the coupled storage and compute architecture. That is, after adding or removing Write Nodes, data can reach balance faster among Write Nodes. The reason is that the Write Node stores all the data in S3 and only needs to store a small amount of data locally at runtime. The scaling up and down process is essentially the migration of Region Peers among nodes. To migrate a Region Peer from one Write Node to another Write Node for management, the target Write node only needs to download a small amount of metadata about this Region from the latest files uploaded to S3 by the Region Peer's original Write Node, and then synchronize the latest Region updates from TiKV to catch up with the Region Leader's progress.
```
Comment (hfxsd) [inline tiflash/tiflash-disaggregated-and-s3.md:26]:
```suggestion
    The Write Node has a faster scaling speed than the TiFlash node in the coupled storage and compute architecture. That is, after adding or removing Write Nodes, data can reach balance faster among Write Nodes. The reason is that the Write Node stores all the data in S3 and only needs to store a small amount of data locally at runtime. The scaling up and down process is essentially the migration of Region Peers among nodes. To migrate a Region Peer from one Write Node to another for management, the target Write node only needs to download a small amount of metadata about this Region from the latest files uploaded to S3 by the Region Peer's original Write Node, and then synchronizes the latest Region updates from TiKV to catch up with the Region Leader's progress. This completes the migration of Region Peers.
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    The Compute Node executes query requests sent from a TiDB node. It first accesses a Write Node to obtain data snapshots, and then reads the latest data (that is, the data not been uploaded to S3 yet) from the Write Node and most of the remaining data from S3.
+
+    The Compute Node uses local disks (usually NVMe SSDs) as a cache for data files to avoid repeatedly reading the same data from remote locations (Write Nodes or S3) and improve query performance.
+
+    The Compute Node is stateless and its scaling speed is at a second level. You can use this feature to reduce costs as follows:
+
+    - When the query workload is low, reduce the number of Compute Nodes to save costs. When there are no queries, you can even stop all Compute Nodes.
+    - When the query workload increases, quickly increase the number of Compute Nodes to ensure query performance.
+
+## Use cases
```
Comment (hfxsd) [inline tiflash/tiflash-disaggregated-and-s3.md:39]:
```suggestion
## Scenarios
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+          storage.remote.cache.capacity: 858993459200           # 800 GiB
+      - host: 172.31.9.2
+        config:
+          flash.disaggregated_mode: tiflash_compute             # This is a Compute Node
+          storage.s3.endpoint: http://s3.{region}.amazonaws.com # S3 endpoint address
+          storage.s3.bucket: mybucket                           # TiFlash stores all data in this bucket
+          storage.s3.root: /cluster1_data                       # Root directory where data is stored in the S3 bucket
+          storage.s3.access_key_id: {ACCESS_KEY_ID}             # Access S3 with ACCESS_KEY_ID
+          storage.s3.secret_access_key: {SECRET_ACCESS_KEY}     # Access S3 with SECRET_ACCESS_KEY
+          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Compute Node, the same configuration way as that of the coupled storage and compute architecture
```
Comment (hfxsd) [inline tiflash/tiflash-disaggregated-and-s3.md:148]:
```suggestion
          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Compute Node. Configure it in the same way as the configuration of the coupled storage and compute architecture
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+          storage.remote.cache.dir: /data1/tiflash/cache        # Local data cache directory of the Compute Node
+          storage.remote.cache.capacity: 858993459200           # 800 GiB
+    ```
+
+    * Note that the above `ACCESS_KEY_ID` and `SECRET_ACCESS_KEY` are directly written in the configuration file. You can also choose to configure them separately using environment variables. If both ways are configured. the priority of environment variables is higher than that of configuration files.
```
Comment (hfxsd) [inline tiflash/tiflash-disaggregated-and-s3.md:153]:
```suggestion
    * Note that the above `ACCESS_KEY_ID` and `SECRET_ACCESS_KEY` are directly written in the configuration file. You can also choose to configure them separately using environment variables. If both ways are configured, the environment variables have higher priority.
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+        tiup cluster reload mycluster -R tidb
+        ```
+
+## Restrictions
+
+- TiFlash does not support in-place switching between the **disaggregated storage and compute architecture** and the **coupled storage and compute architecture**. Before switching architectures, you must remove all existing TiFlash nodes.
+- After the migration from one architecture to another, all TiFlash data needs to be replicated again.
+- Only TiFlash nodes with the same architecture are allowed in the same TiDB cluster. Two architectures cannot coexist in one cluster.
+- The disaggregated storage and compute architecture only supports object storage using the S3 API, while the coupled storage and compute architecture only supports local storage.
+- When using S3 storage, TiFlash nodes cannot obtain the keys of files not on their own nodes, so the [Encryption at Rest](/encryption-at-rest.md) feature is cannot be used.
```
Comment (hfxsd) [inline tiflash/tiflash-disaggregated-and-s3.md:202]:
```suggestion
- When using S3 storage, TiFlash nodes cannot obtain the keys of files not on their own nodes, so the [Encryption at Rest](/encryption-at-rest.md) feature cannot be used.
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+      # 172.31.8.1~2 are TiFlash Write Nodes
+      - host: 172.31.8.1
+        config:
+          flash.disaggregated_mode: tiflash_write               # This is a Write Node
+          storage.s3.endpoint: http://s3.{region}.amazonaws.com # S3 endpoint address
+          storage.s3.bucket: mybucket                           # TiFlash stores all data in this bucket
+          storage.s3.root: /cluster1_data                       # Root directory where data is stored in the S3 bucket
+          storage.s3.access_key_id: {ACCESS_KEY_ID}             # Access S3 with ACCESS_KEY_ID
+          storage.s3.secret_access_key: {SECRET_ACCESS_KEY}     # Access S3 with SECRET_ACCESS_KEY
+          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Write Node, the same configuration way as that of the coupled storage and compute architecture
```
Comment (qiancai) [inline tiflash/tiflash-disaggregated-and-s3.md:117]:
```suggestion
          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Write Node. Configure it in the same way as the directory configuration of the coupled storage and compute architecture
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+      - host: 172.31.8.2
+        config:
+          flash.disaggregated_mode: tiflash_write               # This is a Write Node
+          storage.s3.endpoint: http://s3.{region}.amazonaws.com # S3 endpoint address
+          storage.s3.bucket: mybucket                           # TiFlash stores all data in this bucket
+          storage.s3.root: /cluster1_data                       # Root directory where data is stored in the S3 bucket
+          storage.s3.access_key_id: {ACCESS_KEY_ID}             # Access S3 with ACCESS_KEY_ID
+          storage.s3.secret_access_key: {SECRET_ACCESS_KEY}     # Access S3 with SECRET_ACCESS_KEY
+          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Write Node, the same configuration way as that of the coupled storage and compute architecture
```
Comment (qiancai) [inline tiflash/tiflash-disaggregated-and-s3.md:126]:
```suggestion
          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Write Node. Configure it in the same way as the directory configuration of the coupled storage and compute architecture
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+      # 172.31.9.1~2 are TiFlash Compute Nodes
+      - host: 172.31.9.1
+        config:
+          flash.disaggregated_mode: tiflash_compute             # This is a Compute Node
+          storage.s3.endpoint: http://s3.{region}.amazonaws.com # S3 endpoint address
+          storage.s3.bucket: mybucket                           # TiFlash stores all data in this bucket
+          storage.s3.root: /cluster1_data                       # Root directory where data is stored in the S3 bucket
+          storage.s3.access_key_id: {ACCESS_KEY_ID}             # Access S3 with ACCESS_KEY_ID
+          storage.s3.secret_access_key: {SECRET_ACCESS_KEY}     # Access S3 with SECRET_ACCESS_KEY
+          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Compute Node, the same configuration way as that of the coupled storage and compute architecture
```
Comment (qiancai) [inline tiflash/tiflash-disaggregated-and-s3.md:137]:
```suggestion
          storage.main.dir: ["/data1/tiflash/data"]             # Local data directory of the Compute Node. Configure it in the same way as the directory configuration of the coupled storage and compute architecture
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+    The Write Node uses local disks (usually NVMe SSDs) to cache the latest written data to avoid excessive use of memory.
+
+    The Write Node has a faster scaling speed than the TiFlash node in the coupled storage and compute architecture. That is, after adding or removing Write Nodes, data can reach balance faster among Write Nodes. The reason is that the Write Node stores all the data in S3 and only needs to store a small amount of data locally at runtime. The scaling up and down process is essentially the migration of Region Peers among nodes. To migrate a Region Peer from one Write Node to another for management, the target Write node only needs to download a small amount of metadata about this Region from the latest files uploaded to S3 by the Region Peer's original Write Node, and then synchronizes the latest Region updates from TiKV to catch up with the Region Leader's progress. This completes the migration of Region Peers.
```
Comment (qiancai) [inline tiflash/tiflash-disaggregated-and-s3.md:26]:
```suggestion
    The Write Node has a faster scaling speed than the TiFlash node in the coupled storage and compute architecture. That is, after adding or removing Write Nodes, data can reach balance faster among Write Nodes. The reason is that the Write Node stores all the data in S3 and only needs to store a small amount of data locally at runtime. The scaling up and down process is essentially the migration of Region Peers among nodes. When migrating a Region Peer from one Write Node to another for management, the target Write node only needs to download a small amount of metadata about this Region from the latest files uploaded to S3 by the Region Peer's original Write Node, and then synchronize the latest Region updates from TiKV. This catches up with the Region Leader's progress and completes the migration of Region Peers.
```

File: tiflash/tiflash-disaggregated-and-s3.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+
+    The Write Node uses local disks (usually NVMe SSDs) to cache the latest written data to avoid excessive use of memory.
+
+    The Write Node has a faster scaling speed than the TiFlash node in the coupled storage and compute architecture. That is, after adding or removing Write Nodes, data can reach balance faster among Write Nodes. The reason is that the Write Node stores all the data in S3 and only needs to store a small amount of data locally at runtime. The scaling up and down process is essentially the migration of Region Peers among nodes. When migrating a Region Peer from one Write Node to another for management, the target Write node only needs to download a small amount of metadata about the Region from the latest files uploaded to S3 by the Region Peer's original Write Node, and then synchronize the latest Region updates from TiKV. Then, the target Write node can catch up with the Region Leader's progress and complete the migration of the Region Peers.
```
Comment (qiancai) [inline tiflash/tiflash-disaggregated-and-s3.md:26]:
```suggestion
    The Write Node has a faster scaling speed than the TiFlash node in the coupled storage and compute architecture. That is, after adding or removing Write Nodes, data can reach balance faster among Write Nodes. The reason is that the Write Node stores all the data in S3 and only needs to store a small amount of data locally at runtime. The scaling up and down process is essentially the migration of Region Peers among nodes. When migrating a Region Peer from one Write Node to another for management, the target Write node only needs to download a small amount of metadata about the Region from the latest files uploaded to S3 by the Region Peer's original Write Node, and synchronizes the latest Region updates from TiKV. Then, the target Write node can catch up with the Region Leader's progress and complete the migration of the Region Peers.
```

**Pull request merged**